### DISCUSSION TITLE
Tabs, panels, and sidebar feedback

### DISCUSSION THREAD

**Discussion by adrianthedev** [avo-hq/avo#1073] [Feature feedback]
Let's use this space to provide feedback for the Tabs feature.

Post [an issue](https://github.com/avo-hq/avo/issues/new/choose) if something is broken.

**Comment by jony1993** [comment]
Hi @adrianthedev 

First of all thanks for your work :)

Maybe some of my feedback is already working and I have not read it in the documentation or sth:

1. For some resources we want to have everything in tabs. So sth like first tab is "General", second tab is "Subscriptions" etc.
Problem: At the moment only the fields declared in the root will be visible on the index view. So you have to move out all fields which you want to show in the index view

2. After add/edit a has_many within a a tab, it would be great to open that tab when redirecting back. At the moment you have to click on that specific tab again

**Reply by adrianthedev** [reply]
https://github.com/avo-hq/avo/issues/1115 for the second item

**Comment by jony1993** [comment]
Hi @adrianthedev 

We have continued to test the tab feature.
After working a bit with it: It really improves Avo a lot in our use case.

Regarding my previous # 1 Problem: We fixed it as you have described it in the video. But would be great to avoid duplication of fields

And some other minor optimizations:
1. Add hash to url when you click on a tab. Also check on load the hash in the url and load corresponding tab
2. If you have too many tabs, the design is broken. Add at least a horizontal scroll...

**Reply by adrianthedev** [reply]
Hey @jony1993. Apologies for the radio silence. 

> Regarding my previous # 1 Problem: We fixed it as you have described it in the video. But would be great to avoid duplication of fields

I see the duplication issue. I was thinking of adding a `add_to_root`/`view_on: :index` option to a tab. So if you pass that to the tab, it would automatically add those fields to the `Index` view.
I wouldn't add all to the index view by default, but a solution like this one would make it easier.

Ex:

```ruby
tabs do
  tab "General", view_on: :index do
    # Fields here
  end
end
```

> After add/edit a has_many within a a tab, it would be great to open that tab when redirecting back. At the moment you have to click on that specific tab again

I need to think about an easy to implement solution for this. The hash might help that👆.

> Add hash to url when you click on a tab. Also check on load the hash in the url and load corresponding tab

This is not a bad idea. I have to play around with it a bit

> If you have too many tabs, the design is broken. Add at least a horizontal scroll...

Can you make a screen recording showing the "broken" part?

**Reply by jony1993** [reply]
Hi @adrianthedev 

No problem at all  :)

I think adding all fields from one tab to the index view might not be suitable for all cases.
I think a setting per field as now would be the best. Because you might show also fields from another tab on the index view

Regarding the broken part:
<img width="611" alt="image" src="https://user-images.githubusercontent.com/11912368/181050619-b8c13ebf-cbcf-4ae2-8c5d-bac2eca4c2c3.png">

**Reply by adrianthedev** [reply]
Fixed the broken part https://github.com/avo-hq/avo/pull/1105
